unit_tests/test_plugins.py

```diff
@@ -100,16 +100,29 @@ def test_install_do_remove_unlisted(self, mock_get_plugins_to_install, mock_inst
         mock_get_plugins_to_install.return_value = {plugin_name}
         mock_install_plugins.return_value = {plugin_path}
         orig_remove_unlisted_plugins = hookenv.config()["remove-unlisted-plugins"]
+        orig_plugins = hookenv.config()["plugins"]
         try:
             hookenv.config()["remove-unlisted-plugins"] = "yes"
-            unlisted_plugin = os.path.join(paths.PLUGINS, "unlisted.jpi")
-            with open(unlisted_plugin, "w"):
+            hookenv.config()["plugins"] = "plugin listed"
+            unlisted_plugins = []
+            unlisted_plugin_jpi = os.path.join(paths.PLUGINS, "unlisted.jpi")
+            unlisted_plugin_hpi = os.path.join(paths.PLUGINS, "unlisted.hpi")
+            listed_plugin = os.path.join(paths.PLUGINS, "listed.jpi")
+            plugin_path = os.path.join(paths.PLUGINS, "{}.jpi".format(plugin_name))
+            with open(listed_plugin, "w"):
                 pass
+            for unlisted_plugin in unlisted_plugins:
```

Comment:
I don't see anything being added to unlisted_plugins here - how does this get populated?

Reply:
It gets populated on line 52 by:
```
         unlisted_plugins = existing_plugins - installed_plugins
```